UefiCpuPkg/Library/MpInitLib/DxeMpLib.c

```diff
@@ -652,14 +659,15 @@ InitMpGlobalData (
   // TODO: Check EFI_MEMORY_XP bit set or not once it's available in DXE GCD
   //       service.
   //
-  Status = gDS->GetMemorySpaceDescriptor (Address, &MemDesc);
-  if (!EFI_ERROR (Status)) {
-    gDS->SetMemorySpaceAttributes (
-           Address,
-           ApSafeBufferSize,
-           MemDesc.Attributes & (~EFI_MEMORY_XP)
-           );
-  }
+  // MU_CHANGE START: Set RO and remove NX after copying data into buffer
+  // Status = gDS->GetMemorySpaceDescriptor (Address, &MemDesc);
+  // if (!EFI_ERROR (Status)) {
+  //   gDS->SetMemorySpaceAttributes (
+  //          Address,
+  //          ApSafeBufferSize,
+  //          MemDesc.Attributes & (~EFI_MEMORY_XP)
+  //          );
+  // }

```

Comment:
Should the commented out code be deleted or is it valuable to keep around?

Reply:
I keep them around to make integrations a bit easier